hs_communities/views/communities.py

```diff
@@ -0,0 +1,138 @@
+        """
+        for community in communities:
+            if grp.id in [g.id for g in community.member_groups]:
+                return community
+
+    def get_context_data(self, **kwargs):
+        grps_member_of = []
+        u = User.objects.get(pk=self.request.user.id)
+        groups = Group.objects.filter(gaccess__active=True).exclude(name="Hydroshare Author")
+        # for each group set group dynamic attributes
+        for g in groups:
+            g.is_user_member = u in g.gaccess.members
+            if g.is_user_member:
+                grps_member_of.append(g)
+            g.join_request_waiting_owner_action = g.gaccess.group_membership_requests.filter(request_from=u).exists()
+            g.join_request_waiting_user_action = g.gaccess.group_membership_requests.filter(invitation_to=u).exists()
+            g.join_request = None
+            if g.join_request_waiting_owner_action or g.join_request_waiting_user_action:
+                g.join_request = g.gaccess.group_membership_requests.filter(request_from=u).first() or \
+                                 g.gaccess.group_membership_requests.filter(invitation_to=u).first()
+
+        comms_member_of = [self.group_to_community(g, Community.objects.all()) for g in grps_member_of]
+        return {
+            'communities_list': [c for c in comms_member_of if c is not None]
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class TopicsView(TemplateView):
+    """
+    action: CREATE, READ, UPDATE, DELETE
+    """
+
+    def get(self, request, *args, **kwargs):
+        return render(request, 'pages/topics.html', {'topics_json': self.get_topics_data()})
+
+    def post(self, request, *args, **kwargs):
+        if request.POST.get('action') == 'CREATE':
+            try:
+                new_topic = Topic()
+                new_topic.name = request.POST.get('name').replace("--", "")
+                new_topic.save()
+            except Exception as e:
+                print("TopicsView error creating new topic {}".format(e))
+        elif request.POST.get('action') == 'UPDATE':
+            try:
+                update_topic = Topic.objects.get(id=request.POST.get('id'))
+                update_topic.name = request.POST.get('name')
+                update_topic.save()
+            except Exception as e:
+                print("TopicsView error updating topic {}".format(e))
+        elif request.POST.get('action') == 'DELETE':
+            try:
+                delete_topic = Topic.objects.get(id=request.POST.get('id'))
+                delete_topic.delete(keep_parents=False)
+            except:
+                print("error")
+        else:
+            print("TopicsView POST action not recognized should be CREATE UPDATE or DELETE")
+
+        return render(request, 'pages/topics.html')
+
+    def get_topics_data(self, **kwargs):
+        topics = Topic.objects.all().values_list('id', 'name', flat=False).order_by('name')
```

Comment:
This is surprising to me.  Why are you ordering by name?  I'd expect it to be ordered by the order field.

Reply:
Addressed in 5d9b4648eb93c6c129b8ae142ade2e40bcb8853b